### ISSUE TITLE
Support venv detection on Windows with mingw Python

### ISSUE THREAD

**Issue by lazka** [pytest-dev/pytest#12544]
#### What's the problem this feature will solve?

Currently pytest checks platform depended paths in a directory to see if it might be a venv: https://github.com/pytest-dev/pytest/blob/0ed2d79457b499119693588b71233db957e972c0/src/_pytest/main.py#L370-L387

This works fine with the official CPython releases, but the mingw Python on Windows uses the Unix layout, which looks like this, despite being on Windows:

```console
$ tree -L 1 .venv
.venv
├── bin
├── include
├── lib
└── pyvenv.cfg
```

This results in pytest trying to collect tests in venv files by default.

#### Describe the solution you'd like

I can see three solutions to this:

1) Check for both bin/Scripts

2) First check for the existence of "pyvenv.cfg", which should be in every venv on every platform

3) Replace the current logic and just check for "pyvenv.cfg". "pyvenv.cfg" was introduced in Python 3.3 with the venv module, so it might be enough to assume it exists in every venv. Maybe I'm missing something though.

**Note:** I'm happy to create a PR for any of the above solutions.

**Comment by RonnyPfannschmidt** [comment]
the initial code was added in b32cfc88daad55f6518fc828db7aa770d4e4c80a about 7  years ago when python 2.6 was still supported

so as far as i'm aware there is no reason not to switch to this much simpler test - a pr would be appreciated

**Comment by bluetech** [comment]
Agreed, as far as I know, checking for pyvenv.cfg is sufficient.

**Comment by zachsnicker** [comment]
Hello!
I'm new to contributing to pytest. Can I take this up?

**Comment by webknjaz** [comment]
@zachsnicker I think it's non-controversial enough, I'll assign you. Make sure to include test coverage for the patch in your pull request.

**Comment by lazka** [comment]
Just to be safe I did some additional research, https://peps.python.org/pep-0405/ defines venvs and states "Thus, a Python virtual environment in its simplest form would consist of nothing more than a copy or symlink of the Python binary accompanied by a pyvenv.cfg file and a site-packages directory.". So it's safe to assume that every venv following the spec will contain a pyvenv.cfg.

For the external virtualenv package pyvenv.cfg was added in v20, released 4+ years ago, which is also in Ubuntu 20.04 for example.

So I think this is fine.

I also had a look at the code already, so I'm happy to review if wanted.
